src/test/infra/lttng/test_records_provider_lttng.py

```diff
@@ -76,8 +84,93 @@ def _format(records, columns):
+            return records
+        mocker.patch.object(
+            RecordsProviderLttng, '_rename_column', side_effect=_rename_column)
+
+        def _format(records, columns):
+            return records
+
+        mocker.patch.object(
+            RecordsProviderLttng, '_format', side_effect=_format)
+
+        lttng_mock = mocker.Mock(spec=Lttng)
+
+        data_model_mock = mocker.Mock(spec=Ros2DataModel)
+
+        lttng_mock.data = data_model_mock
+
+        helper_mock = mocker.Mock(spec=RecordsProviderLttngHelper)
+        mocker.patch('caret_analyze.infra.lttng.records_provider_lttng.RecordsProviderLttngHelper',
+                     return_value=helper_mock)
+
+        callback_objects = (1, None)
+        mocker.patch.object(helper_mock, 'get_subscription_callback_objects',
+                            return_value=callback_objects)
+
+        data_model_srv_mock = mocker.Mock(spec=DataModelService)
+        mocker.patch('caret_analyze.infra.lttng.records_provider_lttng.DataModelService',
+                     return_value=data_model_srv_mock)
+        mocker.patch.object(data_model_srv_mock,
+                            '_get_rmw_handle_from_callback_object', return_value=0)
+
+        source_mock = mocker.Mock(spec=FilteredRecordsSource)
+        mocker.patch('caret_analyze.infra.lttng.records_provider_lttng.FilteredRecordsSource',
+                     return_value=source_mock)
+
+        records_mock = RecordsCppImpl(
+            [
+                RecordCppImpl(
+                    {
+                        COLUMN_NAME.TID: 2,
+                        COLUMN_NAME.RMW_TAKE_TIMESTAMP: 3,
+                        COLUMN_NAME.RMW_SUBSCRIPTION_HANDLE: 4,
+                        COLUMN_NAME.MESSAGE: 5,
+                        COLUMN_NAME.SOURCE_TIMESTAMP: 6,
+                    },
+                )
+            ],
+            [
+                ColumnValue(COLUMN_NAME.TID),
+                ColumnValue(COLUMN_NAME.RMW_TAKE_TIMESTAMP),
+                ColumnValue(COLUMN_NAME.RMW_SUBSCRIPTION_HANDLE),
+                ColumnValue(COLUMN_NAME.MESSAGE),
+                ColumnValue(COLUMN_NAME.SOURCE_TIMESTAMP),
+            ]
+        )
+        source_mock._grouped_rmw_records = [records_mock, ]
+
+        subscription_mock = mocker.Mock(spec=SubscriptionStructValue)
+        provider = RecordsProviderLttng(lttng_mock)
+        records = provider.subscription_take_records(subscription_mock)
+
+        expected = RecordsCppImpl(
+            [
+                RecordCppImpl(
+                    {
+                        COLUMN_NAME.RMW_TAKE_TIMESTAMP: 3,
+                        COLUMN_NAME.SOURCE_TIMESTAMP: 6,
+                    },
+                )
+            ],
+            [
+                ColumnValue(COLUMN_NAME.RMW_TAKE_TIMESTAMP),
+                ColumnValue(COLUMN_NAME.SOURCE_TIMESTAMP)
+            ]
+        )
+
+        records.equals(expected)
```

Comment:
Please add `assert`.

```suggestion
        assert records.equals(expected)
```